src/io/overlapped.cr

```diff
@@ -61,7 +66,23 @@ module IO::Overlapped
     end
 
     removed.times do |i|
-      OverlappedOperation.schedule(overlapped_entries[i].lpOverlapped) { |fiber| yield fiber }
+      entry = overlapped_entries[i]
+
+      # at the moment only `::Process#wait` uses a non-nil completion key; all
+      # I/O operations, including socket ones, do not set this field
+      case completion_key = Pointer(Void).new(entry.lpCompletionKey).as(CompletionKey?)
```

Comment:
I suppose it's probably okay to leverage the fact that unions of a reference type and `Nil` are represented as a simple pointer where the null pointe rexpresses a `nil` value. But it feels just a tiny bit iffy. 
Feel free to ignore this comment, or refactor for a more explicit null pointer check if you think it makes sense.

Reply:
Yeah this is the same assumption in `Atomic` and `Box`. Perhaps these special casts can be placed under something like `Crystal::ABI` later